src/tree/TextureManager.mjs

```diff
@@ -135,8 +160,13 @@ export default class TextureManager {
 
     _freeManagedTextureSource(textureSource) {
         if (textureSource.isLoaded()) {
+            // add VRAM tracking if using the webgl renderer
+            this._updateVramUsage(textureSource, -1);
+
             this._nativeFreeTextureSource(textureSource);
             this._addMemoryUsage(-textureSource.w * textureSource.h);
+
+            this._updateVramUsage(textureSource, -1);
```

Comment:
I think the question is why you're decreasing VRAM usage twice when unused texture sources are deleted: 
https://github.com/rdkcentral/Lightning/blob/9fe55acac6bf9ad357cf7ad7c3bb70318d1c9129/src/tree/TextureManager.mjs#L164
https://github.com/rdkcentral/Lightning/blob/9fe55acac6bf9ad357cf7ad7c3bb70318d1c9129/src/tree/TextureManager.mjs#L169

But the second call to `_updateVramUsage()` will lead to an early exit on: 
https://github.com/rdkcentral/Lightning/blob/9fe55acac6bf9ad357cf7ad7c3bb70318d1c9129/src/tree/TextureManager.mjs#L119

because the native texture has been cleared. So the VRAM usage is correct by the second call to `_updateVramUsage()` can be deleted.

Reply:
If you look at _updateVramUsage, you'll see that it calls addVramUsage(), which tracks VRAM usage in a separate variable. The reason for this is because the existing memory management system only tracks pixels. This is not granular enough for the needs of my employer, so they asked me to submit a PR that tracks VRAM usage at the byte level - specifically to track how much vram is being used by RGB textures as opposed to RGBA textures.

https://github.com/rdkcentral/Lightning/pull/395/commits/5348fbed85fb8ceebc19581a5b49b4a76c9fa95b#diff-59016167af069286a9ec655a8610698a60d72331c142ec49463e6fb1b803729bR403-R409

Reply:
I deleted my original reply because I realized I misread your response. I will remove the duplicate call later today and resubmit the PR.